test/new-e2e/tests/containers/docker_test.go

```diff
@@ -0,0 +1,112 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package containers
+
+import (
+	"context"
+	"fmt"
+	fakeintake "github.com/DataDog/datadog-agent/test/fakeintake/client"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/infra"
+	dockervm "github.com/DataDog/test-infra-definitions/scenarios/aws/dockerVM"
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+	"github.com/stretchr/testify/suite"
+	"os"
+	"testing"
+)
+
+type DockerSuite struct {
+	baseSuite
+	stackName string
+}
+
+func TestDockerSuite(t *testing.T) {
+	stackName := "docker-stack"
+	suite.Run(t, &DockerSuite{stackName: stackName})
+}
+
+func (suite *DockerSuite) SetupSuite() {
+	ctx := context.Background()
+
+	stackConfig := runner.ConfigMap{
+		"ddagent:deploy":     auto.ConfigValue{Value: "true"},
+		"ddagent:fakeintake": auto.ConfigValue{Value: "true"},
+	}
+
+	_, stackOutput, err := infra.GetStackManager().GetStackNoDeleteOnFailure(ctx, suite.stackName, stackConfig, dockervm.Run, false)
+
+	if !suite.Assert().NoError(err) {
+		suite.FailNow(ctx)
+	}
```

Comment:
As we don’t do anything with partially created stacks except deleting them, we can just not keeping it in case of failure.

```suggestion
	_, stackOutput, err := infra.GetStackManager().GetStack(ctx, suite.stackName, stackConfig, dockervm.Run, false)
	suite.Require().NoError(err)
```